proto/cosmos/authz/v1beta1/authz.proto

```diff
@@ -25,3 +25,20 @@ message Grant {
   google.protobuf.Any       authorization = 1 [(cosmos_proto.accepts_interface) = "Authorization"];
   google.protobuf.Timestamp expiration    = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
 }
+
+// GrantQueueItem contains the list of granter-grantee-msgtype triple.
+message GrantQueueItem {
+  // ggm_pairs is a list of GrantStoreKey
+  repeated GrantStoreKey ggm_pairs = 1; 
```

Comment:
What's the guarantee that this won't be arbitrary long?

Reply:
Shall we add a limit for grants per expiration? It is possible for someone to create many grants with the same expiration.

Reply:
We can't have a global limit. Technically, all users should be able to create a grant with the same expiry time. One idea would be to lift granter & grantee up to the key - so the list here will be only the msg type urls.

Reply:
How it is different from the current implementation?  Expired grants size will be same in both implementations.

Reply:
You mean the solution in this PR? 
Today, you create a structure with list of `(granter, grantee, msg type url)` per expiration. This can be arbitrary long. If we lift granter and grantee then the list will be limited by the amount of msg types, which is rather short.

Reply:
How's that a problem? I believe what @aleem1314 is talking about could be an issue. If we have 1000s of authorizations to prune, we might face block gas time issue. I am in support of pruning a max-authz per block to avoid potential block gas limitations.

Reply:
Each time a grant is added / deleted or updated, and it's expiration is same as many other grants, it could create a potential issue to deserialize / iterate and serialize that big list.

On the other hand, when we break down the list, the issue will be only once: to delete many expired grants the first time they reach expiration.